server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java

```diff
@@ -216,11 +234,23 @@ private void validateGetComponentDetailsRequest(GetComponentDetailsRequest reque
         }
     }
 
-    private void authorizeRequest(String componentName) {
-        if (Utils.isEmpty(componentName) || !componentName.startsWith(GREENGRASS_CLI_CLIENT_ID_PREFIX) && !CLI_SERVICE
-                .equals(componentName)) {
+    private void authorizeRequest(Permission permission) {
+        if (Utils.isEmpty(permission.getPrincipal())) {
             throw new UnauthorizedError("Component is not authorized to call CLI APIs");
         }
+        // Allow CLI and CLI clients to call anything
+        if (permission.getPrincipal().startsWith(GREENGRASS_CLI_CLIENT_ID_PREFIX)
```

Comment:
do we do any validation to make sure you can't deploy a componment called `GREENGRASS_CLI_CLIENT_ID_PREFIX + <anything>`?

Reply:
No we don't have any particular logic. Of course this code already exists.

Reply:
Let me address it in a different pr.

Reply:
#158